cw_decred/lib/wallet_addresses.dart

```diff
@@ -26,20 +68,42 @@ class DecredWalletAddresses extends WalletAddresses {
     return nAddr;
   }
 
+  List<AddressInfo> getAddressInfos() {
+    if (addressInfos.containsKey(0)) {
+      return addressInfos[0]!;
+    }
+    return <AddressInfo>[];
+  }
+
   List<String> addresses() {
-    final res = libdcrwallet.addresses(walletInfo.name);
-    final addrs = (json.decode(res) as List<dynamic>).cast<String>();
+    final addrs = List<String>.from(usedAddresses);
+    for (final k in addressesMap.keys) {
+      addrs.add(k);
+    }
     return addrs;
   }
 
-  @override
-  set address(String addr) {}
```

Comment:
Selecting an address from the list of addresses on the receive page doesn't work. I think this needs to be implemented for it to work.

Reply:
I removed it because the overridden method was `set address(String address) => _localAddress = address;` and I expected that would make it work, but doesnt seem so.

Reply:
I think that would only work if you also don't override `get address`.

Reply:
I just tried setting the address returned by address to the thing the set argument but didnt work...

Reply:
I think because the getter doesn’t really retain the local variable. It gets an address from libwallet and only returns the local variable if libwallet doesn’t return an address.

I’m thinking that the following might work:
- don’t update the local variable inside the getter, only update via the setter, i.e. when a user selects a specific address.
- In the getter, if the local variable has a value, return that value. It means the user selected that address to use.
- Return an address from libwallet if the local variable is empty.

Reply:
I'm pretty sure I tried that. Can that scene change while looking at it? The edited names dont change until I come back.